File: ios/Sources/EmojiPicker/Presenter/DefaultEmojiPickerPresenter.swift
import Foundation

/// Placeholder presenter until the system emoji keyboard presentation is implemented (issue #5).
final class DefaultEmojiPickerPresenter: EmojiPickerPresenter {
    func present(presentation: String, completion: @escaping (Result<EmojiPickerResult, EmojiPickerError>) -> Void) {
        completion(.failure(EmojiPickerError(code: ErrorCodes.notImplemented)))
    }
}

File: android/src/main/java/app/independo/capacitoremojipicker/presenter/DefaultEmojiPickerPresenter.java
package app.independo.capacitoremojipicker.presenter;

import app.independo.capacitoremojipicker.core.EmojiPickerCallback;
import app.independo.capacitoremojipicker.core.ErrorCodes;

/** Placeholder presenter until the native AndroidX emoji picker UI is implemented (issue #4). */
public class DefaultEmojiPickerPresenter implements EmojiPickerPresenter {

    @Override
    public void present(String presentation, EmojiPickerCallback callback) {
        callback.onError(ErrorCodes.NOT_IMPLEMENTED);
    }
}
